src/components/Table/StatefulTable.test.jsx

```diff
@@ -24,6 +24,28 @@ describe('stateful table with real reducer', () => {
     expect(screen.queryByDisplayValue('option-B')).not.toBeInTheDocument();
   });
 
+  it('should update aria label', () => {
+    const statefulTable = (
+      <StatefulTable
+        {...initialState}
+        options={{ hasFilter: false, hasRowExpansion: false, hasResize: true }}
+        actions={mockActions}
+      />
+    );
+    render(statefulTable);
+    expect(screen.getAllByLabelText('Sort rows by this header in ascending order')).toBeTruthy();
+    expect(tableColumnsWithAlignment[0].isSortable).toBeTruthy();
+    fireEvent.click(screen.getByRole('button', { name: /string/i }));
+    expect(statefulTable.props.actions.table.onChangeSort).toHaveBeenCalled();
```

Comment:
Instead of accessing the props directly, we should expect the mock to have been called. I think this would be the correct object path:

```suggestion
    expect(mockActions.table.onChangeSort).toHaveBeenCalled();
```